Snakefile

```diff
@@ -0,0 +1,94 @@
+import pandas as pd
+import sourmash
+import os
+
+# create wildcard variables for workflow
+metadata = pd.read_csv("inputs/metadata-paired-mgx-mtx.tsv", sep = "\t") # read in metadata as a pandas dataframe
```

Comment:
Python in Snakefile? Wild stuff!! I'd love it if these can move to their individual files, so they can be linted and typed in the future.

Reply:
Ya!!! that's one of the things that makes snakemake so great, is that it's essentially just python, so you can sprinkle in as much python as you want anywhere that you want it.You know, at the expense of maybe no one ever being able to interpret what you did :) I'm not sure what you mean by move to their individual files? Or rather if you mean what I think you mean, I currently don't have a mental model for how to do that within the snakemake ecosystem for all of the python outside of the rules. Python inside the rules I know how to do. Some of the variables could be added to a config file and controlled that way, but I usually find it lighter weight to do it this way

Reply:
hmm ok I'll have to think about this more. 

link to how config files are used: https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html

this is the only snakemake way that I know of to do what I think you are talking about, but might need more conversation. I'm going to merge this as is for now, but am open to revisiting this approach/swapping it out if we figure out a better solution

Reply:
also as part of this review, do you want me to move the python code to a python script with a CLI that gets called by the rule? or do you think its fine for now?

Reply:
I'll read about the config files thank you! 

And all of this is fine for now, seeing this just sparked some ideas and we can change/improve over time. But if Adair is happy, I am happy.